src/coredump/coredump.c

```diff
@@ -1229,6 +1245,50 @@ static int gather_pid_metadata(struct iovec_wrapper *iovw, Context *context) {
         return save_context(context, iovw);
 }
 
+static int process_kernel_in_namespace(const Context *context, int argc, char **argv) {
+        _cleanup_close_ int pidnsfd = -1, mntnsfd = -1, usernsfd = -1, rootfd = -1;
+        _cleanup_strv_free_ char **arguments = NULL;
+        pid_t child;
+        int r;
+
+        /* Invoke systemd-coredump in the container. We pass the same list of arguments
+         * the kernel passed us but we make sure to translate any sensitive metadata */
+
+        log_debug("Invoking %s in the container...", argv[0]);
+
+        /* Copy the argument list now because namespace_fork() is going to replace child's
+         * argv[] completely */
+        arguments = strv_copy(argv);
+        if (!arguments)
+                return log_oom();
+
+        r = namespace_open(context->pid, &pidnsfd, &mntnsfd, NULL, &usernsfd, &rootfd);
+        if (r < 0)
+                return log_error_errno(r, "Can't retrieve namespaces of crashing process: %m");
+
+        r = namespace_fork("(sd-coredumpns)", "(sd-coredump)", NULL, 0,
+                           FORK_RESET_SIGNALS|FORK_DEATHSIG,
+                           pidnsfd, mntnsfd, -1, usernsfd, rootfd, &child);
+        if (r < 0)
+                return r;
+        if (r == 0) {
+                /* Make sure to not leak PID as seen in the host pid namespace */
+                arguments[META_ARGV_PID] = arguments[META_ARGV_NS_PID];
+
+                (void) execv(arguments[0], arguments);
+
+                log_error_errno(errno, "Failed to execute '%s' in container: %m", arguments[0]);
+                _exit(EXIT_FAILURE);
```

Comment:
also, instead of EXIT_FAILURE I'd probably `_exit(255)` here... under the assumption that _exit(EXIT_FAILURE) is something the invoked process might want to use, while 255 might be something the invoking process might use